model/src/main/scala/com/lynbrookrobotics/potassium/model/simulations/SimulatedTwoSidedHardware.scala

```diff
@@ -112,22 +129,31 @@ class SimulatedTwoSidedHardware(constantFriction: Force,
 
   val position = XYPosition(turnPosition.map(a => Degrees(90) - a), forwardPosition)
 
-  val zippedPositions = forwardPosition.zip(turnPosition).zip(position)
-
-  val historyStream = zippedPositions.zip(forwardVelocity).zip(turnVelocity).zipWithTime.map {
-    case (((((fPos, tPos), pos), fVel), tVel), time) =>
-      MomentInHistory(
-        time = time,
+  val zippedStuff: Stream[(((((((Length, Angle), Point), Velocity), AngularVelocity), Dimensionless), Dimensionless), Time)] = forwardPosition.zip(turnPosition)
```

Comment:
Maybe not include the explicit type? I feel that makes it look a lot scarier than it actually is. Your call though.